File: src/app/models/Livre.ts
import { EmitAndSemanticDiagnosticsBuilderProgram } from "typescript";
import { Auteur } from "./Auteur";
import { Categorie } from "./Categorie";
import { Editeur } from "./Editeur";
import { Edition } from "./Edition";
import { Exemplaire } from "./Exemplaire";
import { FileHandle } from "./FileHandle";
import { Motcle } from "./MotCle";
import {PrixLitteraire} from "./PrixLitteraire";
export class Livre{
    public  livre_id!:Number;

    public titre!:String;

    public resume!:String;
    
    public  isbn!:Number;
   
    public categorie_nom!:Categorie["nomCategorie"];
   
    public  motcle_id!:Motcle["id"];

    public auteurs:Auteur[]=[new Auteur(),new Auteur()];

    public prixLiterraire!:PrixLitteraire;

    public  exemplaires !:Exemplaire[];

    public edition: Edition[]=[new Edition()];

    public livreImages:FileHandle[]=[new FileHandle()];

    
}

File: src/main/java/com/medkin/bib/entity/Livre.java
package com.medkin.bib.entity;

import com.fasterxml.jackson.annotation.JsonIgnoreProperties;
import lombok.AllArgsConstructor;
import lombok.Data;
import lombok.NoArgsConstructor;
import org.hibernate.annotations.GenericGenerator;

import javax.persistence.*;
import java.io.Serializable;
import java.util.ArrayList;
import java.util.List;
import java.util.Set;

@Data
@AllArgsConstructor @NoArgsConstructor
@Entity
public class Livre implements Serializable {

    @Id  //primary key
    @GeneratedValue(strategy = GenerationType.IDENTITY)//strategy for generating auto generated number
    @Column(name= "LIVRE_ID", nullable = false)
    Long id;
    @Column
    String titre;
    @Column
    String resume;
    @ManyToMany(fetch=FetchType.LAZY,cascade=CascadeType.ALL)
    @JoinTable(name="LIVRE_IMAGE",
            joinColumns = {
                    @JoinColumn(name = "LIVRE_ID")
            },
            inverseJoinColumns = {
                    @JoinColumn(name = "IMAGE_ID")
            }
    )
    private Set<ImageModel> livreImages;

    @JsonIgnoreProperties({"hibernateLazyInitializer", "handler"})
    @ManyToMany(fetch = FetchType.EAGER, cascade = CascadeType.ALL)
    @JoinTable(name = "LIVRE_EDITION",
            joinColumns = {
                    @JoinColumn(name = "LIVRE_ID")
            },
            inverseJoinColumns = {
                    @JoinColumn(name = "EDITION_ID")
            }
    )
    private List<Edition> edition=new ArrayList<>();

    @JsonIgnoreProperties({"hibernateLazyInitializer", "handler"})
    @ManyToMany(fetch = FetchType.LAZY, cascade = CascadeType.ALL)
    @JoinTable(name = "LIVRE_AUTEUR",
            joinColumns = {
                    @JoinColumn(name = "LIVRE_ID")
            },
            inverseJoinColumns = {
                    @JoinColumn(name = "AUTEUR_ID")
            }
    )
    private List<Auteur> auteurs=new ArrayList<>();

    @JsonIgnoreProperties({"hibernateLazyInitializer", "handler"})
    @OneToOne(fetch = FetchType.LAZY, cascade = CascadeType.ALL)
    @JoinTable(name = "LIVRE_MOTCLE",
            joinColumns = {
                    @JoinColumn(name = "LIVRE_ID")
            },
            inverseJoinColumns = {
                    @JoinColumn(name = "MOTCLE_ID")
            }
    )
    private MotCle motCle;

    @JsonIgnoreProperties({"hibernateLazyInitializer", "handler"})
    @OneToMany(fetch = FetchType.LAZY, cascade = CascadeType.ALL)
    @JoinTable(name = "LIVRE_PRIXLITTERAIRE",
            joinColumns = {
                    @JoinColumn(name = "LIVRE_ID")
            },
            inverseJoinColumns = {
                    @JoinColumn(name = "PRIXLITTERAIRE_ID")
            }
    )
    private List<PrixLitteraire> prixLitteraires=new ArrayList<>();


}
